File: CodeClash/Application/Dtos/UserSolutionStatsDto.cs
namespace Application.Dtos;

public class UserSolutionStatsDto
{
    public int TotalCount { get; set; }
    public int SuccessCount { get; set; }
    public int FailedCount { get; set; }
    public double SuccessRate { get; set; }

    public DateTime? FirstSubmissionDate { get; set; }
    public DateTime? LastSubmissionDate { get; set; }
    public int ActiveDays { get; set; }

    public Dictionary<string, int> AttemptsByDate { get; set; } = new();
}

File: frontend-react/src/interfaces/api/core/UserSolutionStatsDto.ts
export default interface UserSolutionStatsDto {
  totalCount: number;
  successCount: number;
  failedCount: number;
  successRate: number;
  firstSubmissionDate: string | null;
  lastSubmissionDate: string | null;
  activeDays: number;
  attemptsByDate: Record<string, number>;
}